test/workers/delete_plain_object_worker_test.rb

```diff
@@ -54,4 +54,31 @@ def test_perform_destroy_without_association
       refute object.destroyed?
     end
   end
+
+  class StaleObjectErrorTest < DeletePlainObjectWorkerTest
+    def setup
+      @service = FactoryGirl.create(:simple_service)
+      FactoryGirl.create(:simple_service, account: @service.account) # This way the previous service can be destroyed
+      @proxy = FactoryGirl.create(:proxy, service: @service)
+    end
+
+    def test_perform_on_father_after_child_destroyed
+      service1 = @service
+      @service.proxy
```

Comment:
This is the _equivalent_ (simpler) of:
https://github.com/3scale/porta/blob/bb8ab7f2f498b92d6e3ddb8a2e58df5884d60d32/app/workers/delete_object_hierarchy_worker.rb#L73-L80

Reply:
I think it does not happens in this code but really but due to a race condition. Because the deletion of the service and the proxy happens at the same time in different workers.

1. The first worker will load the proxy through `dependent: :destroy` 
2. The second worker just deletes it
3. The first worker tries to to delete the proxy it but it is already gone so it raises ActiveRecord::StaleObjectError

Testing race condition is hard, it involves fiber and hooking into some rails internals ... So I agree that it is kinda of equivalent of the rails internal one. (Will try to find a way to test it, good exercise and we can be used in the future)

Reply:
aha, nice discovery, thank you! 👍 💪